src/workerd/server/server.c++

```diff
@@ -2538,8 +2538,12 @@ kj::Promise<void> Server::listenOnSockets(config::Config::Reader config,
     auto rewriter = kj::heap<HttpRewriter>(httpOptions, headerTableBuilder);
 
     tasks.add(listener
-        .then([this, &service, rewriter = kj::mv(rewriter), physicalProtocol]
+        .then([this, &service, rewriter = kj::mv(rewriter), physicalProtocol, name]
               (kj::Own<kj::ConnectionReceiver> listener) mutable {
+      KJ_IF_MAYBE(stream, controlOverride) {
+        auto message = kj::str("{\"event\":\"listen\",\"socket\":\"", name, "\",\"port\":", listener->getPort(), "}\n");
```

Comment:
Hmm I guess we're inventing an ad hoc protocol here. Makes me a little nervous, but I guess using capnp rpc feels like overkill, so I'm not sure I have any better ideas.